crates/cli/src/datadog_utils.rs

```diff
@@ -57,42 +58,68 @@ pub fn get_datadog_variable_value(variable: &str) -> anyhow::Result<String> {
     Err(anyhow!("cannot find variable DD_{}", variable))
 }
 
-// if we put the first argument to true staging, override the value and use staging.
-// otherwise, use the DD_SITE variable or the default site
-fn get_datadog_site(use_staging: bool) -> String {
-    if use_staging {
-        STAGING_DATADOG_SITE.to_string()
+// If a DD_HOSTNAME envvar has been specified, use it; otherwise, use the staging hostname
+// if use_staging is true; otherwise, use the DD_SITE envvar with 'api.' prepended;
+// otherwise, use the default hostname.
+fn get_datadog_hostname(use_staging: bool) -> String {
+    if let Ok(hostname) = get_datadog_variable_value("HOSTNAME") {
+        hostname
+    } else if use_staging {
+        STAGING_DATADOG_HOSTNAME.to_string()
+    } else if let Ok(site) = get_datadog_variable_value("SITE") {
+        format!("api.{}", site).to_string()
     } else {
-        get_datadog_variable_value("SITE").unwrap_or(DEFAULT_DATADOG_SITE.to_string())
+        DEFAULT_DATADOG_HOSTNAME.to_string()
     }
 }
 
-// get rules from one ruleset at datadog
-// it connects to the API using the DD_SITE, DD_APP_KEY and DD_API_KEY and retrieve
-// the rulesets. We then extract all the rulesets
-pub fn get_ruleset(ruleset_name: &str, use_staging: bool) -> Result<RuleSet> {
-    let site = get_datadog_site(use_staging);
-    let app_key = get_datadog_variable_value("APP_KEY");
-    let api_key = get_datadog_variable_value("API_KEY");
-
-    let url = format!(
-        "https://api.{}/api/v2/static-analysis/rulesets/{}?include_tests=false&include_testing_rules=true",
-        site, ruleset_name
-    );
+#[derive(PartialEq)]
+enum UseKeys {
+    // Do not add APP/API keys or a JWT token to the request.
+    DoNotUse,
+    // Add APP/API keys or JWT token to the request, if available.
+    NotRequired,
+    // Add APP/API keys or JWT token to the request, and fail if not available.
+    Required,
+}
```

Comment:
Why this? We should always use the keys when possible as it prevents rate-limits. Anonymous request are all accounted for the same rate-limit (and can lead to rate-limiting issues). When we use an api/app key, the rate limit is calculated for the account. For this reason, requests should be made authenticated if a token is available.

Reply:
I do this to preserve the current behavior of the functions I'm replacing. `get_ruleset` uses the keys if they are available but works without them; `get_default_rulesets_name_for_language` does not use the keys at all; `get_diff_aware_information` requires the keys.

I can change `get_default_rulesets_name_for_language` to pass the keys if available, but that's not its current behavior.

Reply:
The part that I do not understand is why we need to have `DoNotUse`? We should use whatever keys we have when we have it.

Reply:
My PR keeps the current behavior, in which one of the functions does not include keys. If the current behavior is wrong, I can fix it in the same PR and remove `DoNotUse`.